apps/omg_watcher_info/test/support/factory.ex

```diff
@@ -58,12 +58,69 @@ defmodule OMG.WatcherInfo.Factory do
 
   alias OMG.WatcherInfo.DB
 
+  @eth OMG.Eth.RootChain.eth_pseudo_address()
+
+  @doc """
+  Block factory.
+
+  Generates a block in an incremental blknum of 1000, 2000, 3000, etc.
+  """
   def block_factory() do
     %DB.Block{
       blknum: sequence(:block_blknum, fn seq -> seq * 1000 end),
-      hash: sequence(:block_hash, fn seq -> <<seq::256>> end),
+      hash: insecure_random_bytes(32),
       eth_height: sequence(:block_eth_height, fn seq -> seq end),
       timestamp: sequence(:block_timestamp, fn seq -> seq * 1_000_000 end)
     }
   end
+
+  @doc """
+  Transaction factory.
+
+  Generates a transaction without any transaction output and no associated block.
+
+  To generate a transaction with closest data to production, consider associating the transaction
+  to a block and generating transaction outputs associated with this transaction.
+  """
+  def transaction_factory() do
+    %DB.Transaction{
+      txhash: sequence(:transaction_hash, fn seq -> <<seq::256>> end),
+      txindex: 0,
+      txbytes: insecure_random_bytes(32),
+      sent_at: DateTime.utc_now(),
+      metadata: insecure_random_bytes(32),
+      block: nil,
+      inputs: [],
+      outputs: []
+    }
+  end
```

Comment:
same as above. please avoid random test data and remove use of `insecure_random_bytes(32)` and `DateTime.utc_now()`

```suggestion
      def transaction_factory(attrs \\ nil) do
        # a default 'sent_at' DateTime to use if none is passed in via args
        {:ok, default_test_datetime, 0} = DateTime.from_iso8601("2019-12-12T11:11:11Z")

        sent_at = attrs[:sent_at] || default_test_datetime

        transaction = %OMG.WatcherInfo.DB.Transaction{
          txhash: sequence(:transaction_hash, fn seq -> <<seq::256>> end),
          txindex: sequence(:transaction_txindex, fn seq -> seq end),
          txbytes: sequence(:transaction_txbytes, fn seq -> <<seq::256>> end),
          sent_at: sequence(:transaction_sent_at, fn seq -> DateTime.add(sent_at, seq) end),
          metadata: sequence(:transaction_metadata, fn seq -> <<seq::256>> end),
          block: attrs[:block] || build(:block)
        }

        merge_attributes(transaction, attrs)
```

Reply:
Disagreeing here. I think random test data and DateTime is fine.